### DISCUSSION TITLE
(general) unreliable bot pathing with docking ports.

### DISCUSSION THREAD

**Discussion by Ek-Videogames** [FakeFishGames/Barotrauma#14770] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The method of bot pathfinding between docked hulls is very sensitive to the exact distance between the last 'inside' node and the next exterior node; usually on the end of the ladder.

If this is even slightly off bots will often refuse to even enter the next hull, even if hatches are automatically open or there are no hatches at all. The exact spacing between hatches - and the resulting size of the "auto generated" hull produced between the ports varies between different combinations of sub + shuttle, sub + station, sub + beacon, sub + sub, etc. As such it's kind of impossible to account for every possibility on the end of the sub-builder.

- Interestingly, in some cases I've noted that AI is capable of crossing the hull threshold in one direction on their own but be unable to path back the other way.

AI pathing needs to be just a little bit more robust to reliably deal with these slight variations without breaking. 

Perhaps if AI is considered "stuck", force their node target to the next valid node (that being the next node on their path likely the one which intersects the docked hull) or force them to target the nearest node or nearest INSIDE node of the docked hull's navigation mesh.

The main issue always seems to be that bots are either unable to recognize the docked connection: "I can't reach that location",
or bots are able to recognize the docked connection but become stuck targeting the last interior node of the hull and not properly making the 'transition' navigation hand-off process to switch to using the new hull's navigation to enter.

### Reproduction steps

Nothing to preview

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Shmeckle** [comment]
I am experiencing this exact bug currently, to further expand on this though, based on observations I have discovered that the bots are able to cross the hull threshold in some reliable and predictable situations.

Firstly, this overall issue appears to be far less prevalent/common in sub editor as opposed to a live game for whatever reason.

Secondly, as a tested example, I have a custom made ship and a custom made drone. Whenever I set a bot to repair electrical devices, it will enter the drone, located below the main ship with no issue and pathfind and repair the object consistently. However, it will not be able to then leave the drone, and cannot even grab the ladder to begin ascending, no matter the changes I make to waypoints IF the bot is then defaulting back to keeping the main ship reactor on. However, if the bot repairs something in the drone and then tries to path to a junction next to the reactor, it can path back up just fine.

These kinds of pathing issues at first appear semi random, but upon closer inspection, seem to be reliable to replicate in certain strange configurations. I do not know if this will help fix the issue, but I can also provide my ship with drone if that would help to fix the bug.

In multiplayer, this issue isn't a big deal at all, in singleplayer, it can be the difference between your AI crew successfully holding off an attack, or having crucial maintenance personal getting softlocked, causing the entire mission to fail while you are overboard. It's extremely frustrating.
